tools/src/test/java/cz/o2/proxima/tools/groovy/ConsoleTest.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2017-2021 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.tools.groovy;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import cz.o2.proxima.repository.AttributeDescriptor;
+import cz.o2.proxima.repository.Repository;
+import cz.o2.proxima.storage.StreamElement;
+import cz.o2.proxima.storage.commitlog.Position;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+/** Test {@link cz.o2.proxima.tools.groovy.Console} and various aspects of Closure/classloading. */
+public class ConsoleTest {
+
+  private final Config config = ConfigFactory.load("test-reference.conf").resolve();
+  private final Repository repo = Repository.ofTest(config);
+
+  @Test(timeout = 20000)
```

Comment:
what takes so long here?

Reply:
It generates and compiles the Environment class.